machine/v1/stable.controlplanemachineset.openstack.testsuite.yaml

```diff
@@ -0,0 +1,277 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
```

Comment:
@ingvagabund I added all the test cases that I thought would be useful. Please let me know what you think.

@JoelSpeed for a quick review on the tests, if their results make sense to you from a CPMS point of view.